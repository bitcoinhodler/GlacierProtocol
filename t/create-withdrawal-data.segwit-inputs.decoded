{
  "txid": "18060d87d55989ced3d0ec538ecd3ac84d6501a26fdd5a572d4629281926766c",
  "hash": "18060d87d55989ced3d0ec538ecd3ac84d6501a26fdd5a572d4629281926766c",
  "version": 2,
  "size": 1358,
  "vsize": 1358,
  "weight": 5432,
  "locktime": 0,
  "vin": [
    {
      "txid": "2042d4fa2ad5e8fb6df4cb03bef9e9833542209ea9d59067ac24ef0ca51c22d3",
      "vout": 1,
      "scriptSig": {
        "asm": "0 304402207766694c32fccc15959a050d60b31aee063d507bd40b706833b6ae6fcfe2ff1e022017e10c87307f72d72b091b6ad1b440e3d24785428164bd6dee7d57f70061e58e[ALL] 3044022028204e8b5a94d417201551fce13859f5c7e2b6903d515ddbc72b2b5f447bbc83022006c8fe0c97628af97acdf44638743eb165ce3da6c275ecf2ed32bdb6c3b0dbd5[ALL] 522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae",
        "hex": "0047304402207766694c32fccc15959a050d60b31aee063d507bd40b706833b6ae6fcfe2ff1e022017e10c87307f72d72b091b6ad1b440e3d24785428164bd6dee7d57f70061e58e01473044022028204e8b5a94d417201551fce13859f5c7e2b6903d515ddbc72b2b5f447bbc83022006c8fe0c97628af97acdf44638743eb165ce3da6c275ecf2ed32bdb6c3b0dbd5014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      },
      "sequence": 4294967295
    },
    {
      "txid": "daf8e3f26d1d0fd2b648fd292bdefc17be6e20240a5f17fca2b828e07b5e1971",
      "vout": 0,
      "scriptSig": {
        "asm": "0 3044022073d0ef43627911ec9e31825e2f8e45b245964446e17d73758ef1742af803f5e80220219409fbae75d02912b1b118cc8d25e7a04d5e5fdb920a44f0098788824a5692[ALL] 3044022028a336f9091d4868b6f2a021c7d3c4d048cd136e389d5ff3b26f54dcf203e04c0220053ff7faa3a613d6c0ff0e956f9184af9e1259eaed1a95eba73786d1f3c7e184[ALL] 522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae",
        "hex": "00473044022073d0ef43627911ec9e31825e2f8e45b245964446e17d73758ef1742af803f5e80220219409fbae75d02912b1b118cc8d25e7a04d5e5fdb920a44f0098788824a569201473044022028a336f9091d4868b6f2a021c7d3c4d048cd136e389d5ff3b26f54dcf203e04c0220053ff7faa3a613d6c0ff0e956f9184af9e1259eaed1a95eba73786d1f3c7e184014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      },
      "sequence": 4294967295
    },
    {
      "txid": "38e7e6f7a03267b5e703969a808f8817b244e29934a8693ee51e4d836efc3db5",
      "vout": 0,
      "scriptSig": {
        "asm": "0 3044022066b297344c72d4d7b0058e1de817a8b5aac32dc8e69f9ede9cfb6e1f94c0a76d02203ab55e319338df76d5df00da8b283abc75c0fd8a1d323c24e504e56db386f999[ALL] 304402207d40d0a5f2654153c71c063fca423f6f456f89b627d92c5e7bf0d52ee6b7f23402203983412e04f9778aa925e10a987258e24923396420105ec3aebac60218fda8ce[ALL] 522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae",
        "hex": "00473044022066b297344c72d4d7b0058e1de817a8b5aac32dc8e69f9ede9cfb6e1f94c0a76d02203ab55e319338df76d5df00da8b283abc75c0fd8a1d323c24e504e56db386f9990147304402207d40d0a5f2654153c71c063fca423f6f456f89b627d92c5e7bf0d52ee6b7f23402203983412e04f9778aa925e10a987258e24923396420105ec3aebac60218fda8ce014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      },
      "sequence": 4294967295
    },
    {
      "txid": "11e61fd2a18d68c8daa095b3c4f80ba698d3d6d2bdedbd59f88b11f224c458fe",
      "vout": 0,
      "scriptSig": {
        "asm": "0 304402202e60c96234b3d2d8cbea81159eef7e3f6d550eeff327a80a8a1b1821ead18904022017caa42db921ad314b748600cfcb7873154645122e693e7004efc5f0f392255a[ALL] 304402201f6a170b1dd719d360f3a96e27cd852fd9ebe6a11b1519afc489717e91c994220220766336e7d79684584c966e7d1f6bc6f528780993e1a6d76484e6032a3d5b853e[ALL] 522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae",
        "hex": "0047304402202e60c96234b3d2d8cbea81159eef7e3f6d550eeff327a80a8a1b1821ead18904022017caa42db921ad314b748600cfcb7873154645122e693e7004efc5f0f392255a0147304402201f6a170b1dd719d360f3a96e27cd852fd9ebe6a11b1519afc489717e91c994220220766336e7d79684584c966e7d1f6bc6f528780993e1a6d76484e6032a3d5b853e014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      },
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 1.69997793,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 8546e747c8094376b0a0af44bc5bbb8dd4f0218d OP_EQUAL",
        "hex": "a9148546e747c8094376b0a0af44bc5bbb8dd4f0218d87",
        "reqSigs": 1,
        "type": "scripthash",
        "addresses": [
          "2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow"
        ]
      }
    }
  ]
}
