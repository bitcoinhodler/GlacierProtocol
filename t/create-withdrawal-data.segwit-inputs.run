#!/bin/bash
set -e

# Test withdrawing from transactions that were created by SegWit wallets.
# I have four transactions to try:
# 1. Electrum p2wpkh-in-p2sh with change
# 2. Electrum native p2wpkh with change
# 3. Electrum p2wpkh-in-p2sh with no change
# 4. Electrum native p2wpkh with no change


$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT 2>&1
y
y
y
y
y
y
2NAVkgszAvjZZwWpHpbCaxSaNqCCXPD8Bts
522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae
2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow
4
02000000000101b5d5a5c7f516548efbaac687114573004ecaf3c227e01b715407e85bb84a596201000000171600140f30187d8f55baa6d3051459fd8e5c6635413ae4ffffffff02107e88020000000017a9144af6d047bbb0999be644a054a562e8fbce68c90387a07f88020000000017a914bd3996bb9a6c401da7e65d4dbb3a9b978cd045ec870247304402200c3d4aa6f9964a13e6b90b577f806923c3783dfed1381bf52aa0b1693ab5a9e20220054cbe00b6194cbe47dc4b48431a8ca378076626e8df8a34bf8b97cdc2bed47a012102c692df13e3749aeb65867935fe6ac4b2209de5af343e33026de10dfe28c7021a00000000
02000000000101205d369f87f7961fa6b4d2ac90c88b798d3350ede3322e38c1571201d38c8fb30100000000ffffffff02084b87020000000017a914bd3996bb9a6c401da7e65d4dbb3a9b978cd045ec8770b3890200000000160014dbe2b5b174943a51c77906152e174cbadbfa221102473044022035c5a9b2fe363b0c828cb04b25e02ed16bbd96174790312110b08b0b7766eae002202c829c5af56e2a9f36878c1f7cb875334fb8f19da1b08eea467ac70327b31a0b012103b4b0846aa01f17b5fc9bbe5f47a74fec60d95c41841ee3f8b33cfcc1d3071ea500000000
02000000000101b8750ab1693891727f99c251e3b1282731628bd2690abfd2421d6a86ccdb61c7010000001716001444fa86792b21b82401b5c161d234467f67d52a9dffffffff01887d88020000000017a914bd3996bb9a6c401da7e65d4dbb3a9b978cd045ec8702473044022007f92ce19d79a6c50b4ce2e362adc281b50d4fb793ca4eee909e4e43c0c214150220152deed6df0ab105a54f8521fdf2962452ca6b8482bda9ea2d4c01dbb1bc7abd012102675b4ac4f6e2e25eb56f00a4ca284a59efdb3ccad279a1a33d68c9784c56aa3600000000
02000000000101882378f62ecaa3b7ac5e94479fcdb5795f4899d2dba22287d9eac550479036620100000000ffffffff01ffb289020000000017a914bd3996bb9a6c401da7e65d4dbb3a9b978cd045ec870247304402200822c12f52e0a11671c2f90c0b5af55281b12bf43f0c9b9f0a78e9134609e06b022067e8e47c32f2d7fcc737828621fe2a741e4ee3b4754c1143e03e12d86354b7c7012103ce631e50d09126256d0369a79b7d9b5c856683720687249a745daee589c78e3300000000
2
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4
1
y

y
INPUT
