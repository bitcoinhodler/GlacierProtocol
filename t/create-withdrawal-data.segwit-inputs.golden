Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Please paste raw transaction #2 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Please paste raw transaction #3 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Please paste raw transaction #4 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 1.69999151 BTC

How many private keys will you be signing this transaction with? 
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00001390 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00001390, you have 1.69997761 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to 2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow (leave blank to withdraw all funds stored in these unspent transactions): With no change output, the transaction fee is reduced, and 1.69997793 BTC will be sent to your destination.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

1.69999151 BTC in unspent supplied transactions
1.69997793 BTC going to destination address 2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow
Fee amount: 0.00001358

Signing with private keys: 
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4


Confirm? (y/n): 
Calculating transaction...


Sufficient private keys to execute transaction?
True

Raw signed transaction (hex):
020000000448121218e5e85b910bdfb12be923b84e5bd1b7bda6dfdf188762b415695cc97501000000fd1e0100473044022049e993d3818ab6b631ffa4fbcbd99a9c712317d8c6e323699b2be2e22f85f180022061d9b807ac4ab2c809f95f30e1224a5964d2f47923eaaf94407d1ff50848112f01473044022079492bcbda7858e80880cec52757958a52b307980b0ecc21eb1825d0e0c9565602203332b762cd164be76d1ec964726d5fc9ff1fd8a922c9be34ccbc0c16bbcd95b7014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354aeffffffffdf1e32efe8b3222707f4d796fcf2a282195642a1e9da8fd0e510113acc97cc9200000000fd1e0100473044022079c9887ae3051740a88b53ec48ec6d34e72b7a42724c2fa20d3a1b96965048a60220284828fd8a4e9b943051ee001be33beb39fca31cd440040fff9a58bf78714564014730440220286d55ae014535b4e2521142fc0e4dec74eacb8f3ad37f164b495f79413b32870220233430050f097d1559465d946e31034667239d922c0e40974104dfe74e6641fa014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354aeffffffffbf710bfffa3c2faba4b1f4e366ae12de6ff6ecc21d4b2aed36bd78aa3e1dc8a600000000fd1e0100473044022047cfb42b7c7f9f1500973553e9826be7dbcb700aa2794d4627f44bbf597ae0bd02204cdf82a74f32531f849d489737d61ffcb49f87b9b27fedb51973b52b849463bf01473044022008ecc1f558f7b7fd734a0bbb0a497ed8ab9047c592b546629a40063c8cd67d1102204f226915d9efa144eec808375bc0a35df640cef247fda176d05848e425cc87a5014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354aeffffffffa2bfdfe27c66b657352f963931a6587f7d453a2e05cfe2633e5a3e43a4a8986400000000fd1e0100473044022044499bd6b6cdaaa583634722c3431f3ad16c167ba145ad70f1ebfd1626531b24022075a3de7bec8f70ad332ccb4c620222aeb0fdd04f76cdcfaaeaa5bc564a91f884014730440220767d2e87adfa7d95055222d004798ee95acc747fc87131af4718dd8f366c64ac02207d857a4b20f1b3976321674e77c79fccf131b48637fa81955ec1bd3181587e5e014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354aeffffffff01e1f5210a0000000017a9148546e747c8094376b0a0af44bc5bbb8dd4f0218d8700000000

Transaction fingerprint (md5):
ca63db94948541157fb39f97f357e464
QR code for transaction written to transaction.png
